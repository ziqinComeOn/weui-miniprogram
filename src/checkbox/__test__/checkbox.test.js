import path from 'path'
import simulate from 'miniprogram-simulate'

describe('checkbox-group & checkbox', () => {
    let id

    beforeAll(() => {
        id = simulate.load(path.resolve(__dirname, './index'), { less: true })
    })

    test('basic checkbox', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))
        expect(container.toJSON()).toMatchSnapshot()
    })

    test('radio value', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))

        const radioGroup = container.querySelector('.radio-group')

        radioGroup.dispatchEvent('change', {
            detail: {
                value: '0'
            }
        })
        await simulate.sleep(10)
        expect(container.data.radioValue).toBe('0')

        radioGroup.dispatchEvent('change', {
            detail: {
                value: '1'
            }
        })
        await simulate.sleep(10)
        expect(container.data.radioValue).toBe('1')
    })

    test('checkbox value', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))

        const checkboxGroup = container.querySelector('.checkbox-group')

        checkboxGroup.dispatchEvent('change', {
            detail: {
                value: ['0', '1']
            }
        })
        await simulate.sleep(10)
        expect(container.data.checkboxValue).toBe('0,1')

        checkboxGroup.dispatchEvent('change', {
            detail: {
                value: ['1']
            }
        })
        await simulate.sleep(10)
        expect(container.data.checkboxValue).toBe('1')
    })
})
