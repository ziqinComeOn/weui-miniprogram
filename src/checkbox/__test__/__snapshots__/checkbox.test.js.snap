// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox-group & checkbox basic checkbox 1`] = `
<main>
  <mp-cells
    title="单选列表项"
  >
    <mp-checkbox-group
      class="radio-group"
      bind:change="radioChange"
    >
      <radio-group
        class=""
        bind:change="checkedChange"
      >
        <mp-checkbox
          class="test-radio"
        >
          <mp-cell
            bind:tap="checkedChange"
          >
            <wx-view
              class="weui-cell   weui-check__label   ^weui-cell_radio "
              hoverClass=""
              bind:tap="navigateTo"
            >
              <wx-view
                class="weui-cell__bd "
              >
                <wx-view>
                  cell standard
                </wx-view>
              </wx-view>
              <wx-view
                class="weui-cell__ft "
              >
                <wx-view
                  slot="footer"
                >
                  <wx-radio
                    checked=""
                    class="weui-check"
                    color=""
                    disabled=""
                    value="0"
                  />
                  <wx-icon
                    class="weui-icon-checked"
                  />
                </wx-view>
              </wx-view>
            </wx-view>
          </mp-cell>
        </mp-checkbox>
        <mp-checkbox
          class="test-radio"
        >
          <mp-cell
            bind:tap="checkedChange"
          >
            <wx-view
              class="weui-cell   weui-check__label weui-cell_wxss  ^weui-cell_radio "
              hoverClass=""
              bind:tap="navigateTo"
            >
              <wx-view
                class="weui-cell__bd "
              >
                <wx-view>
                  cell standard
                </wx-view>
              </wx-view>
              <wx-view
                class="weui-cell__ft "
              >
                <wx-view
                  slot="footer"
                >
                  <wx-radio
                    checked=""
                    class="weui-check"
                    color=""
                    disabled=""
                    value="1"
                  />
                  <wx-icon
                    class="weui-icon-checked"
                  />
                </wx-view>
              </wx-view>
            </wx-view>
          </mp-cell>
        </mp-checkbox>
      </radio-group>
    </mp-checkbox-group>
  </mp-cells>
  <mp-cells
    title="复选列表项"
  >
    <mp-checkbox-group
      class="checkbox-group"
      bind:change="checkboxChange"
    >
      <checkbox-group
        class=""
        bind:change="checkedChange"
      >
        <mp-checkbox
          class="test-checkbox"
        >
          <mp-cell
            bind:tap="checkedChange"
          >
            <wx-view
              class="weui-cell   weui-check__label   ^weui-cell_checkbox "
              hoverClass=""
              bind:tap="navigateTo"
            >
              <wx-view
                class="weui-cell__hd "
              >
                <wx-view
                  slot="icon"
                >
                  <wx-checkbox
                    checked=""
                    class="weui-check"
                    color=""
                    disabled=""
                    value="0"
                  />
                  <wx-icon
                    class="weui-icon-checked"
                  />
                </wx-view>
              </wx-view>
              <wx-view
                class="weui-cell__bd "
              >
                <wx-view>
                  standard is dealt for u.
                </wx-view>
              </wx-view>
            </wx-view>
          </mp-cell>
        </mp-checkbox>
        <mp-checkbox
          class="test-checkbox"
        >
          <mp-cell
            bind:tap="checkedChange"
          >
            <wx-view
              class="weui-cell   weui-check__label weui-cell_wxss  ^weui-cell_checkbox "
              hoverClass=""
              bind:tap="navigateTo"
            >
              <wx-view
                class="weui-cell__hd "
              >
                <wx-view
                  slot="icon"
                >
                  <wx-checkbox
                    checked=""
                    class="weui-check"
                    color=""
                    disabled=""
                    value="1"
                  />
                  <wx-icon
                    class="weui-icon-checked"
                  />
                </wx-view>
              </wx-view>
              <wx-view
                class="weui-cell__bd "
              >
                <wx-view>
                  standard is dealicient for u.
                </wx-view>
              </wx-view>
            </wx-view>
          </mp-cell>
        </mp-checkbox>
      </checkbox-group>
      <radio-group
        class=""
        bind:change="checkedChange"
      />
    </mp-checkbox-group>
  </mp-cells>
</main>
`;
